ndk/src/media/media_codec.rs

```diff
@@ -246,9 +313,121 @@ impl MediaCodec {
             inner: NonNull::new(unsafe {
                 ffi::AMediaCodec_createEncoderByType(c_string.as_ptr())
             })?,
+            async_notify_callback: None,
         })
     }
 
+    #[cfg(feature = "api-level-28")]
+    pub fn set_async_notify_callback(
```

Comment:
Do you want to deal with this along with documenting the other methods, since you offered to do so? :)

Reply:
Maybe, maybe not :)

Reply:
Added docs. Is the formatting ok? I have omitted the comments about `AMediaCodec_getInputBuffers()` and `AMediaCodec_getOutputBuffers()` since these functions seem not to exist, and the comments about `userdata` since this is hidden in the Rust API.